integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientLoanCreditBalanceRefundandRepaymentTypeIntegrationTest.java

```diff
@@ -438,4 +439,73 @@ public void newGoodwillCreditCreatesCorrectJournalEntriesForCashAccountingTest()
 
     }
 
+    @Test
```

Comment:
Please add test cases to cover the adjusting of the goodwill credit, payout refund and merchant issued refund and chargeback is not possible and they throw the proper error.